src/states/game.gd

```diff
@@ -296,12 +299,37 @@ func spawn_arrow(id: String, power: float) -> void:
 	new_arrow.transform = player_head.get_global_transform()
 	var shot_speed := BASE_SHOT_SPEED + (MAX_SHOT_SPEED - BASE_SHOT_SPEED) * power
 	new_arrow.velocity = shot_speed * -player_head.get_global_transform().basis.z.normalized()
+	new_arrow.spawn_pickup.connect(self.arrow_pickup_middleman)
 	arrows.add_child(new_arrow)
 	if arrows.get_child_count() > MAX_ARROWS_LOADED:
 		arrows.get_child(0).queue_free()
 	new_arrow.archer.shooting_sound()
 
 
+func arrow_pickup_middleman(spawn_transform: Transform3D) -> void:
```

Comment:
Rename to something like `on_arrow_pickup_spawn`, since it's connecting from a signal.